### PR TITLE
Add ziya CPU example

### PR DESCRIPTION
**Pull request by ivy-lv11**
Add ziya model and the performance could be found in the issue https://github.com/analytics-zoo/nano/issues/1033

### CODE REVIEW

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# ziya
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md:1]: ziya -> Ziya

Comment (Oscilloscope98) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md:1]: Same for other places

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#### 2.4 Sample Output
+#### [IDEA-CCNL/Ziya-Coding-34B-v1.0](https://huggingface.co/IDEA-CCNL/Ziya-Coding-34B-v1.0)
+```log
+Inference time: xxxx s
+-------------------- Prompt --------------------
+Question: What is AI?
+
+ Answer:
+-------------------- Output --------------------
+Question:  What is AI?
```
Comment (plusbang) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md:70]: The sample output seems mismatch `ZIYA_PROMPT_FORMAT`, please confirm it.

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+export OMP_NUM_THREADS=48
+numactl -C 0-47 -m 0 python ./generate.py --prompt 'What is AI？'
+```
+More information about arguments can be found in [Arguments Info](#23-arguments-info) section. The expected output can be found in [Sample Output](#24-sample-output) section.
+
+#### 2.3 Arguments Info
+In the example, several arguments can be passed to satisfy your requirements:
+
+- `--repo-id-or-model-path`: str, argument defining the huggingface repo id for the ziya model to be downloaded, or the path to the huggingface checkpoint folder. It is default to be `'IDEA-CCNL/Ziya-Coding-34B-v1.0'`.
+- `--prompt`: str, argument defining the prompt to be inferred (with integrated prompt format for chat). It is default to be `What is AI?`.
```
Comment (plusbang) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md:58]: According to model card, this model supports Chinese. I feel it maybe better to use Chinese prompt by default.

**Comment by Oscilloscope98** [comment]
Please also rebase this PR on main branch :)

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md (hunk 1 above)
Comment (ivy-lv11) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md:1]: Sure, I will fix them.

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md (hunk 2 above)
Comment (ivy-lv11) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md:70]: Ok, and I will confirm it.

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md (hunk 3 above)
Comment (ivy-lv11) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/README.md:58]: Though the model supports Chinese, we utilize the coding capacities of ziya-coding-34B better when we use the prompt in code format like `def func(arr):`. So I set the default prompt `def quick_sort(arr):\n`

File: python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/generate.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    parser.add_argument('--prompt', type=str, default="def quick_sort(arr):\n",
+                        help='Prompt to infer') 
+    parser.add_argument('--n-predict', type=int, default=128,
+                        help='Max tokens to predict')
+
+    args = parser.parse_args()
+    model_path = args.repo_id_or_model_path
+
+    
+    from bigdl.llm.transformers import AutoModelForCausalLM
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/generate.py:42]: Please add some comments to show users that use_cache=True is important to be set. Let we do before:
```python
# enabling `use_cache=True` allows the model to utilize the previous
# key/values attentions to speed up decoding;
# to obtain optimal performance with BigDL-LLM INT4 optimizations,
# it is important to set use_cache=True for Ziya models
```

Comment (ivy-lv11) [reply python/llm/example/CPU/HF-Transformers-AutoModels/Model/ziya/generate.py:42]: Sure, I will add them in the code to remind users.

File: python/llm/example/CPU/PyTorch-Models/Model/ziya/README.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+### 1. Install
+We suggest using conda to manage the Python environment. For more information about conda installation, please refer to [here](https://docs.conda.io/en/latest/miniconda.html#).
+
+After installing conda, create a Python environment for BigDL-LLM:
+```bash
+conda create -n llm python=3.9 # recommend to use Python 3.9
+conda activate llm
+
+pip install --pre --upgrade bigdl-llm[all] # install the latest bigdl-llm nightly build with 'all' option
+
```
Comment (Oscilloscope98) [inline python/llm/example/CPU/PyTorch-Models/Model/ziya/README.md:18]: Missing of einops here?

**Review by Oscilloscope98 — APPROVED** [review]
LGTM

**Pull request merged**